src/main/java/org/opensearch/plugin/insights/rules/model/Attribute.java

```diff
@@ -118,7 +119,13 @@ public static void writeValueTo(StreamOutput out, Object attributeValue) throws
         if (attributeValue instanceof List) {
             out.writeList((List<? extends Writeable>) attributeValue);
         } else if (attributeValue instanceof SearchSourceBuilder) {
-            ((SearchSourceBuilder) attributeValue).writeTo(out);
+            if (out.getVersion().onOrAfter(Version.V_3_4_0)) {
```

Comment:
Don't forget to change all version checks to 3.5 since 3.4 deadline has passed https://opensearch.org/releases/